app/src/rgb_underglow.c

```diff
@@ -187,20 +215,35 @@ static int zmk_rgb_underglow_init(struct device *_arg)
 	}
 
     state = (struct rgb_underglow_state){
-        hue: 0,
-        saturation: 100,
-        brightness: 100,
-        animation_speed: 3,
-        current_effect: 0,
+        hue: CONFIG_ZMK_RGB_UNDERGLOW_HUE_START,
+        saturation: CONFIG_ZMK_RGB_UNDERGLOW_SAT_START,
+        brightness: CONFIG_ZMK_RGB_UNDERGLOW_BRT_START,
+        animation_speed: CONFIG_ZMK_RGB_UNDERGLOW_SPD_START,
+        current_effect: CONFIG_ZMK_RGB_UNDERGLOW_EFF_START,
         animation_step: 0,
+#ifdef CONFIG_ZMK_RGB_UNDERGLOW_ON_START
         on: true
+#else
+        on: false
+#endif
     };
 
-    k_timer_start(&underglow_tick, K_NO_WAIT, K_MSEC(50));
+    settings_subsys_init();
+    settings_register(&rgb_conf);
+    settings_load();
```

Comment:
Somewhat the same thing with the `subsys_init`, this should probably be in a more generic file that waits for other pieces to initialize (and register their configs), and then one `settings_load` should be called at once.